api/files/serializers.py

```diff
@@ -370,6 +370,13 @@ def get_file_guid(self, obj):
                 return guid._id
         return None
 
+    def get_file_guid_or_error(self, obj):
+        if obj:
+            guid = obj.get_guid()
+            if guid:
+                return guid._id
+        raise JSONAPIException
```

Comment:
@adlius , I suspect this would break file serializer when file don't have a guid, the further fix would be hide this relationship when a file doesn't have a guid, making a another fix

Reply:
Fixed and improved in https://github.com/CenterForOpenScience/osf.io/pull/10532